src/components/hmi_message_handler/src/mqueue_adapter.cc

```diff
@@ -70,8 +72,13 @@ class ReceiverThreadDelegate : public threads::ThreadDelegate {
     }
   }
 
+  void exitThreadMain() OVERRIDE {
+    shutdown_requested_ = true;
```

Comment:
@vlantonov,
How we stop thread if we are waiting something with `mq_receive`?

Without `pthread_cancel` we need to somehow cancel mq reading.

I see 2 possible solutions:
* Using mqueue with `poll` - it's working on Linux, possible works on QNX.
 * See [TCP example](https://github.com/smartdevicelink/sdl_core/blob/c0107a52570b4b189dcc8aa5a055a1f38ae91c81/src/components/transport_manager/src/transport_adapter/threaded_socket_connection.cc#L209)
*  Using `O_NONBLOCK ` reading and checking result flag